include/zephyr/drivers/interrupt_controller/riscv_plic.h

```diff
@@ -12,6 +12,15 @@
 #ifndef ZEPHYR_INCLUDE_DRIVERS_RISCV_PLIC_H_
 #define ZEPHYR_INCLUDE_DRIVERS_RISCV_PLIC_H_
 
+#include <zephyr/device.h>
+
+/**
+ * @brief For configuring IRQ on each individual PLIC device.
+ *
+ * @param dev PLIC device instance.
+ */
+typedef void (*riscv_plic_irq_config_func_t)(const struct device *dev);
```

Comment:
why this is in a generic public interface? Is it going to be used by users?

Reply:
not at the moment, and not sure if it's ever gonna be used by users, just following the pattern of [uart.h](https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/drivers/uart.h#L153)

Reply:
had another look, seem rather unnecessary, removed